backends/ebpf/ebpfBackend.cpp

```diff
@@ -73,7 +73,10 @@ void run_ebpf_backend(const EbpfOptions& options, const IR::ToplevelBlock* tople
 
     Target* target;
     if (options.target.isNullOrEmpty() || options.target == "kernel") {
-        target = new KernelSamplesTarget(options.emitTraceMessages);
+        if (!options.generateToXDP)
```

Comment:
I think the main reason is that we treat XDP as kind of optimization of the default TC-based design, since our design in XDP doesn't implement full PSA spec (see README - no CoS, no CLONE_E2E, no packet recirculation). 

But to be honest, I don't have a strong opinion here. We could probably make it a new target pretty easily but then we would require some refactoring of other targets (e.g., `kernel` target is not a good name anymore since we have more than 1 kernel-based target, also how to resolve conflicts between PSA-eBPF and p4c-xdp?).